src/app/storage/FabricTableImpl.hpp

```diff
@@ -0,0 +1,824 @@
+    using TypedFabricEntryData = FabricEntryData<StorageId, StorageData, Serializer::kPersistentStorageDataBufferMax(),
+                                                 Serializer::kPersistentFabricBufferMax(), Serializer::kMaxPerFabric()>;
+
+    VerifyOrReturnError(IsInitialized(), CHIP_ERROR_INTERNAL);
+    TypedFabricEntryData fabric(mEndpointId, fabric_index, mMaxPerFabric, mMaxPerEndpoint);
+
+    ReturnErrorOnFailure(fabric.Load(mStorage));
+
+    return fabric.RemoveEntry(mStorage, entry_id);
+}
+
+/// @brief This function is meant to provide a way to empty the entry table without knowing any specific entry Id. Outside of this
+/// specific use case, RemoveTableEntry should be used.
+/// @param fabric_index Fabric in which the entry belongs
+/// @param entry_idx Position in the Table
+/// @return CHIP_NO_ERROR if removal was successful, errors if failed to remove the entry or to update the fabric after removing it
+template <class StorageId, class StorageData, size_t kIteratorsMax>
+CHIP_ERROR FabricTableImpl<StorageId, StorageData, kIteratorsMax>::RemoveTableEntryAtPosition(EndpointId endpoint,
+                                                                                              FabricIndex fabric_index,
+                                                                                              EntryIndex entry_idx)
+{
+    using Serializer           = DefaultSerializer<StorageId, StorageData>;
+    using TypedFabricEntryData = FabricEntryData<StorageId, StorageData, Serializer::kPersistentStorageDataBufferMax(),
+                                                 Serializer::kPersistentFabricBufferMax(), Serializer::kMaxPerFabric()>;
+    using TypedTableEntryData  = TableEntryData<StorageId, StorageData, Serializer::kPersistentStorageDataBufferMax()>;
+
+    VerifyOrReturnError(IsInitialized(), CHIP_ERROR_INTERNAL);
+
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    TypedFabricEntryData fabric(endpoint, fabric_index, mMaxPerFabric, mMaxPerEndpoint);
+    TypedTableEntryData entry(endpoint, fabric_index, entry_idx);
+
+    ReturnErrorOnFailure(fabric.Load(mStorage));
+    err = entry.Load(mStorage);
+    VerifyOrReturnValue(CHIP_ERROR_NOT_FOUND != err, CHIP_NO_ERROR);
+    ReturnErrorOnFailure(err);
+
+    return fabric.RemoveEntry(mStorage, entry.mStorageId);
+}
+
+template <class StorageId, class StorageData, size_t kIteratorsMax>
+CHIP_ERROR FabricTableImpl<StorageId, StorageData, kIteratorsMax>::RemoveFabric(FabricIndex fabric_index)
+{
+    using Serializer           = DefaultSerializer<StorageId, StorageData>;
+    using TypedFabricEntryData = FabricEntryData<StorageId, StorageData, Serializer::kPersistentStorageDataBufferMax(),
+                                                 Serializer::kPersistentFabricBufferMax(), Serializer::kMaxPerFabric()>;
+
+    VerifyOrReturnError(IsInitialized(), CHIP_ERROR_INTERNAL);
+
+    for (auto endpoint : app::EnabledEndpointsWithServerCluster(kEntryEndpointClusterID()))
```

Comment:
Ah, this is where clusters come in?

Couldn't this just go over all endpoints, period, and not have to worry about being tied to a specific single cluster ID?

Reply:
If I am guessing what my past self was thinking, I think the goal was to limit KVS read attempts. Now looking at this, I thing aside from bridges the endpoint number count will likely never be high enough to make a significant time difference.

Reply:
Updated to not tie to single cluster ID